lib/redis_memo/memoize_query/cached_select.rb

```diff
@@ -209,6 +217,10 @@ def self.with_new_query_context
   #
   # Note: Arel::Nodes#each returns a list in post-order, and it does not step
   # into Union nodes. So we're implementing our own DFS
+  #
+  # @param node [Arel::Nodes::Node]
+  #
+  # @return [RedisMemo::MemoizeQuery::CachedSelect::BindParams]
```

Comment:
this is one of the methods that i think will be valuable to explain the DFS. but i figured that the code itself actually served better than text explanation. I think a graph representation will be nice if we want to explain more vividly! 

(but probably does not  matter as people will not use much of this method)

Reply:
Should we mark these methods that we don't expect people to use as private?

Reply:
i think this can be done in a diff PR, currently, i think there might be some other methods can be private, which need to be adjusted